### ISSUE TITLE
Undici failures

### ISSUE THREAD

**Issue by targos** [nodejs/citgm#1051]
With https://github.com/nodejs/node/pull/52002:

https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/3398/nodes=alpine-latest-x64/testReport/junit/(root)/citgm/undici_v6_7_0/
https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/3398/nodes=rhel8-x64/testReport/junit/(root)/citgm/undici_v6_7_0/

@nodejs/undici

**Comment by targos** [comment]
Is it? The run was done after the revert.

**Comment by Uzlopak** [comment]
Does the citgm connect via debug port?

**Comment by KhafraDev** [comment]
```
 test at test/node-test/debug.js:54:1
  debug#fetch (346.571414ms)
   AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
   
   4 !== 6
   
       at res.<computed> [as strictEqual] (/home/iojs/tmp/citgm_tmp/19886eeb-1a0b-44d6-951c-284ce1a45418/undici/node_modules/@matteo.collina/tspl/tspl.js:52:35)
       at Socket.<anonymous> (/home/iojs/tmp/citgm_tmp/19886eeb-1a0b-44d6-951c-284ce1a45418/undici/test/node-test/debug.js:79:12)
       at Socket.emit (node:events:531:35)
       at endReadableNT (node:internal/streams/readable:1696:12)
       at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
     generatedMessage: true,
     code: 'ERR_ASSERTION',
     actual: 4,
     expected: 6,
     operator: 'strictEqual'
   }
 test at test/node-test/debug.js:89:1
  debug#undici (186.572836ms)
   AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
   
   21 !== 6
   
       at res.<computed> [as strictEqual] (/home/iojs/tmp/citgm_tmp/19886eeb-1a0b-44d6-951c-284ce1a45418/undici/node_modules/@matteo.collina/tspl/tspl.js:52:35)
       at Socket.<anonymous> (/home/iojs/tmp/citgm_tmp/19886eeb-1a0b-44d6-951c-284ce1a45418/undici/test/node-test/debug.js:117:12)
       at Socket.emit (node:events:531:35)
       at endReadableNT (node:internal/streams/readable:1696:12)
       at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
     generatedMessage: true,
     code: 'ERR_ASSERTION',
     actual: 21,
     expected: 6,
     operator: 'strictEqual'
   }
 test at test/node-test/autoselectfamily.js:1:1
  test/node-test/autoselectfamily.js (30000.124732ms)
   'test timed out after 30000ms'
```

related to https://github.com/nodejs/undici/issues/2600

**Comment by targos** [comment]
Ok, so we have to wait for the next libuv update.

**Comment by KhafraDev** [comment]
it looked like a different error to me, this test has had quite a few failures cc @metcoder95

**Comment by Uzlopak** [comment]
yeah, it is an annoying file

**Comment by metcoder95** [comment]
Ok... that's new, the mismatch on the assertions is weird; I'll try to revisit this and possibly rewrite the testing scenarios, haven't had the time yet.

If this is a blocker of some sort, feel free to skip them for now

**Comment by Uzlopak** [comment]
Should now work as we patched the failing test.

**Comment by targos** [comment]
https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/240/

**Comment by targos** [comment]
On OSX:

````
09:22:16 error:                     | ✖ failing tests:                                                                          
09:22:16 error:                     |                                                                                           
09:22:16 error:                     | test at test/issue-803.js:1:1                                                             
09:22:16 error:                     | ✖ test/issue-803.js (30147.778213ms)                                                      
09:22:16 error:                     | 'test timed out after 30000ms'                                                            
09:22:16 error:                     |                               
````

**Comment by Uzlopak** [comment]
added macos-latest to our ci

https://github.com/nodejs/undici/pull/2952

**Comment by mcollina** [comment]
https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/241/

**Comment by KhafraDev** [comment]
```
 test at test/issue-803.js:1:1
  test/issue-803.js (30101.051783ms)
   'test timed out after 30000ms'
```

@Uzlopak

**Comment by Uzlopak** [comment]
What should I do? The test has afaik already a timeout of 60000ms. So failing after 30000 ms, makes no sense.

**Comment by mcollina** [comment]
Just skip the test if citgm.

**Comment by Uzlopak** [comment]
Is there a env variable set in case of citgm?

**Comment by targos** [comment]
I don't think you can specifically detect cigtm, but you can add custom env variables using the `envVar` key in the lookup.

Example: https://github.com/nodejs/citgm/blob/12e6902b7a90c42edf60a1f4aa0ef9c19efdfddc/lib/lookup.json#L245-L252

**Comment by mcollina** [comment]
I've released v6.19.3 that should address some of this
